storage/aptosdb/src/state_store/mod.rs

```diff
@@ -549,15 +551,17 @@ impl StateStore {
     }
 
     pub fn reset(&self) {
-        let (buffered_state, smt_ancestors) = Self::create_buffered_state_from_latest_snapshot(
+        let lock = self.reset_lock();
+        lock.reset();
+    }
+
+    pub fn reset_lock(&self) -> ResetLock<'_> {
```

Comment:
Another cherry-pick from #28; it looks like a needless refactoring now, but will come useful later.